src/files.js

```diff
@@ -90,6 +92,18 @@ async function doUpload(cozy, data, method, path, options) {
     }
   }
 
+  if (sourceAccount) {
+    finalpath += `${
+      finalpath.includes('?') ? '&' : '?'
+    }SourceAccount=${sourceAccount}`
+  }
+
+  if (sourceAccountIdentifier) {
+    finalpath += `${
+      finalpath.includes('?') ? '&' : '?'
+    }SourceAccountIdentifier=${sourceAccountIdentifier}`
```

Comment:
Maybe you can extract a small helper function that adds a parameter to the query-string in `finalpath`. It will avoid to repeat those 3 lines several times.

Reply:
I considered it but found it more readable like this finally. Since the options and the query string parameters are not exactly the same, we have to pass to many parameters.

But maybe you are right. With a good naming, it may be better.